compiler/rustc_trait_selection/src/traits/coherence.rs

```diff
@@ -297,7 +297,12 @@ fn implicit_negative<'cx, 'tcx>(
             predicate: p,
         })
         .chain(obligations)
-        .find(|o| !selcx.predicate_may_hold_fatal(o));
+        .find(|o| {
+            selcx.evaluate_root_obligation(o).map_or(
+                true, // Overflow has occurred, and treat it as a possible obligation.
```

Comment:
This should be false, you moved the `!`

Reply:
I just want to confirm. Are you confident that's the case? 

 If `﻿evaluate_root_obligation` encounters an overflow, it will return ﻿`Err`. In that case, we should return ﻿`true` because we are looking for the failing obligation.